functions/src/common/context.ts

```
@@ -36,6 +38,19 @@ export function getDatastore(): Datastore {
   return datastore;
 }
 
+export async function getMailService(): Promise<MailService | undefined> {
+  if (!mailService) {
+    try {
+      const config = await MailService.gerMailServerConfig(getDatastore());
+
+      mailService = new MailService(config);
+    } catch (e) {
+      console.error(e);
```

Why swallow the exception?